### PR TITLE
Add tests for the Navigation and the Preference features

### PR DESCRIPTION
**Pull request by Shayan105**
# Added Tests:
### NavigationActionsTest.kt:
- navigateToTopLevelDestinationClearsBackStack()
- navigateToTopLevelDestinationClearsBackStackWithAUTH_PATH()

### PreferenceScreenTest.kt:
- savingPreferencesCallsGoBack()
- savingPreferencesCallsUpdatePreferences()

### PreferencesRepositoryFirestoreTest.kt:
- documentSnapshotToPreferences_ReturnsDefaultPreferencesIfNotFoundOnFireStore()
- documentSnapshotToPreferences_throwsErrorIfPrefsIsNull()

# Summary:
This PR mainly aims at improving the line coverage of the project.

### CODE REVIEW

**Review by AlecLeGrooker — COMMENTED** [review]
Very good work overall ! It is great that you were able to improve coverage. 
And congrats for this test navigateToTopLevelDestinationClearsBackStackWithAUTH_PATH() to be sure that once logged out, one can not access anything, I would not have thought about it.

File: app/src/androidTest/java/com/android/sample/screen/PreferencesScreenTest.kt [hunk 1]
```diff
@@ -69,4 +75,31 @@ class PreferencesScreenTest {
...
   }
+
+  @Test
+  fun savingPreferencesCallsUpdatePreferences() {
+    composeTestRule.setContent { PreferencesScreen(mockNavHostController, preferencesViewModel) }
+    val secondPreferences = Preferences(unitsSystem = UnitsSystem.IMPERIAL, weight = WeightUnit.LBS)
+
+    // Simulate user changing the system of units and weight unit
+    // from (METRIC, KG) to (IMPERIAL, LBS)
+    composeTestRule.onNodeWithTag("unitsSystemButton").performClick()
```
[the remark this answers is not in the document]
Comment (Shayan105) [reply app/src/androidTest/java/com/android/sample/screen/PreferencesScreenTest.kt:86]: It is already tested by an existing test : testDropdownMenuSelectionTextUpdate. But yes Sure I can add this verification !

File: app/src/androidTest/java/com/android/sample/screen/PreferencesScreenTest.kt [hunk 2]
```diff
@@ -69,4 +75,31 @@ class PreferencesScreenTest {
...
+    composeTestRule.onNodeWithTag("weightUnitButton").performClick()
+    composeTestRule.onNodeWithTag("weightUnitLBS").performClick()
+
+    // Save the changes
+    composeTestRule.onNodeWithTag("preferencesSaveButton").performClick()
+
+    // check if the preferences were updated in the repository
+    verify(mockPreferencesRepository).updatePreferences(eq(secondPreferences), any(), any())
+  }
+
```
[the remark this answers is not in the document]
Comment (Shayan105) [reply app/src/androidTest/java/com/android/sample/screen/PreferencesScreenTest.kt:97]: Good catch !

File: app/src/test/java/com/android/sample/model/preferences/PreferencesRepositoryFirestoreTest.kt [hunk 3]
```diff
@@ -115,4 +116,27 @@ class PreferencesRepositoryFirestoreTest {
...
   }
+
+  @Test
+  fun documentSnapshotToPreferences_ReturnsDefaultPreferencesIfNotFoundOnFireStore() {
+    val mockDocumentSnapshot = mock(DocumentSnapshot::class.java)
+    `when`(mockDocumentSnapshot.exists()).thenReturn(false)
+
+    val result = preferencesRepositoryFirestore.documentSnapshotToPreferences(mockDocumentSnapshot)
+
+    assert(result == PreferencesViewModel.defaultPreferences)
```
[the remark this answers is not in the document]
Comment (Shayan105) [reply app/src/test/java/com/android/sample/model/preferences/PreferencesRepositoryFirestoreTest.kt]: indeed. Solved now

**Comment by Shayan105** [comment]
Review taken into account and changes have been made.

**Review by AlecLeGrooker — COMMENTED** [review]
Perfect everything looks fine to me so you can merge to main.

**Review by AlecLeGrooker — APPROVED** [review]
You can merge.

**Pull request merged**